files/default/neutron-ha-tool.py

```diff
@@ -585,8 +589,54 @@ def migrate_router(qclient, router_id, agent, target):
     if router_id not in list_routers_on_l3_agent(qclient, target['id']):
         raise RuntimeError("Failed to add router_id=%s from agent_id=%s" %
                            (router_id, agent['id']))
+    if wait_for_ports:
+        wait_router_ports_active(qclient, router_id, target['host'])
 
 
+def wait_router_ports_active(qclient, router_id, target_host, maxtries=60):
+    """
+    Returns nothing. Waits for all non-distributed ports of a router
+    for being in the ACTIVE status on a specific host
+
+    :param qclient: A neutron client
+    :param router_id: The id of the router to check
+    :param target_host: The host on which the ports should be ACTIVE
+                        (reflected by the binding:host_id attribute of
+                        a port)
+    :param maxtries: Maximum number of times the ports' status is polled,
+                     an expection is raised there are still ports with a
+                     non ACTIVE status after that.
+    """
+
+    # FIXME: Don't check the "network:router_interface_distributed" ports
+    # as those are not hosted on the network node
+    router_port_list = qclient.list_ports(device_id=router_id, fields='id')
```

Comment:
http://developer.openstack.org/api-ref/networking/v2/?expanded=list-ports-detail
https://wiki.openstack.org/wiki/Neutron/APIv2-specification#Filtering_and_Column_Selection

and the neutronclient source code :/

And yes the SDK API doc were not really helpful.